src/WorkerPool.js

```diff
@@ -94,6 +112,21 @@ class PoolWorker {
   onWorkerMessage(message, finalCallback) {
     const { type, id } = message;
     switch (type) {
+      case 'end': {
+        const err = new Error('worker terminated');
+        const jobIds = Object.keys(this.jobs);
+        for (let i = 0; i < jobIds.length; i++) {
```

Comment:
You can use `for (const key of Object.keys(this.jobs)`

Reply:
Done!